### PR TITLE
[RTM] Resource annotation

### PR DESCRIPTION
**Pull request by effigies**
Back-burner work, but we should be specifying resources better. I'm consistently seeing my machine dip into swap while claiming it has 13/14 GiB of memory free. (Thanks to @oesteban for the status updates that make this clear.)

This starts with:

* Standardize interfaces on `num_threads` input argument so that it's automatically synced with `Node.n_procs` (see nipy/nipype#2200)
* Remove explicit setting of `inputs.num_threads`, to let `Node.n_procs` manage the syncing

Todo:

* Add memory annotation to a lot more nodes

### CODE REVIEW

**Comment by oesteban** [comment]
This PR will likely also make the changes necessary to close #712

**Comment by effigies** [comment]
Yeah. And I'm hoping your profiling work can give us good estimates of the memory usage that we can just run through and plug in.

**Comment by oesteban** [comment]
I can confirm lots of MemoryErrors for specific datasets on LS5 (less memory than Stampede2)

**Comment by oesteban** [comment]
I'm brewing a new singularity image with these annotations to run on ls5.

**Comment by oesteban** [comment]
Some nodes might be overestimated, but that should be ok. Let's merge this and see what happens.

File: fmriprep/workflows/util.py [hunk 1]
```diff
@@ -299,7 +298,7 @@ def init_bbreg_wf(use_bbr, bold2t1w_dof, omp_nthreads, name='bbreg_wf'):
     mri_coreg = pe.Node(
         MRICoregRPT(dof=bold2t1w_dof, sep=[4], ftol=0.0001, linmintol=0.01,
                     num_threads=omp_nthreads, generate_report=not use_bbr),
```
Comment (effigies) [inline fmriprep/workflows/util.py]: Can drop `num_threads` here.

File: fmriprep/workflows/bold.py [hunk 2]
```diff
@@ -1285,16 +1285,11 @@ def _aslist(in_value):
...
     ])
 
-<<<<<<< HEAD
-    bold_to_mni_transform = pe.Node(
-        MultiApplyTransforms(interpolation="LanczosWindowedSinc", float=True),
-        name='bold_to_mni_transform', mem_gb=8, n_procs=omp_nthreads)
-    # bold_to_mni_transform.terminal_output = 'file'
-=======
     bold_to_mni_transform = pe.Node(MultiApplyTransforms(
         interpolation="LanczosWindowedSinc", float=True, num_threads=omp_nthreads,
```
Comment (effigies) [inline fmriprep/workflows/bold.py]: This is re-adding num_threads to `bold_to_mni_transform.interface`.

File: fmriprep/workflows/bold.py [hunk 3]
```diff
@@ -1002,10 +1002,11 @@ def init_bold_reg_wf(freesurfer, use_bbr, bold2t1w_dof, bold_file_size_gb, omp_n
             (inputnode, merge_xforms, [('fieldwarp', 'in2')])
         ])
 
-    bold_to_t1w_transform = pe.Node(
-        MultiApplyTransforms(interpolation="LanczosWindowedSinc", float=True),
-        name='bold_to_t1w_transform', mem_gb=8, n_procs=omp_nthreads)
-    # bold_to_t1w_transform.terminal_output = 'file'  # OE: why this?
+    bold_to_t1w_transform = pe.Node(MultiApplyTransforms(
+        interpolation="LanczosWindowedSinc", float=True, copy_dtype=True,
+        num_threads=omp_nthreads),
```
Comment (effigies) [inline fmriprep/workflows/bold.py]: `num_threads` re-added to `bold_to_t1w_transform.interface`.

**Comment by effigies** [comment]
Let me just push a quick fix...

File: fmriprep/workflows/bold.py (hunk 2 above)
Comment (oesteban) [reply fmriprep/workflows/bold.py]: Sorry about that. Lazy merge with badly resolved conflicts.

**Pull request merged**